lib.php

```diff
@@ -81,7 +80,51 @@ public static function has_config() {
 
     public static function get_config_options() {
 
-        // Get default config;
+        $defaultconfig = self::get_default_config();
+
+        $config = array();
+        $config['generalsettings']            = self::define_general_settings($defaultconfig);
+        $config['transfersettings']           = self::define_transfer_settings($defaultconfig);
+        $config['objectfssettingsfilesystem'] = self::define_client_selection($defaultconfig);
+        $clients = self::get_client_components('client');
+
+        foreach ($clients as $name => $classname) {
+            /** @var \module_objectfs\client\object_client $client */
+            $client = new $classname($defaultconfig);
+            $config[$name . 'settings'] = $client->define_settings_form();
```

Comment:
This also has to be changed to account for when no modules are present, currently it throws an error when module_azure is not present